### PR TITLE
Working scim entry get for person

### LINKED ISSUE #2983: REST API: SSH key tags not gettable
Using Kanidm 1.3.0:

Looking at the Swagger docs and Nginx request logs, the APIs for SSH public keys seem to be:

`/v1/person/{name}/_ssh_pubkeys` - returns all SSH public key values as a JSON array
`/v1/person/{name}/_attr/ssh_publickey` - returns all keys as a JSON array of strings formatted as `"<tag>: <value>"`
`/v1/person/{name}/_ssh_pubkeys/{tag}` - returns a single SSH public key value if you know it's "tag".

There appears to be no way to programmatically get a SSH public key's tag.

I would have expected one of `/_ssh_pubkeys` or `/_attr/ssh_publickey` to return a JSON object along the lines of:

```json
{
  "tag1": "type base64 user@host",
  "tag2": "type base64 user@host"
}
```

however that only current result I can get, which then requires string parsing, is:

```json
[
  "tag1: type base64 user@host",
  "tag2: type base64 user@host"
]
```

As an aside, it would also be nice to have an API that just returns the `authorized_keys` file's expected format, without JSON.

### PR DESCRIPTION
**Pull request by Firstyear**
# Change summary

- Add scim person get
- Allow querying to limit the attributes returned

Fixes #2983 

Checklist

- [ x ] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
The only place I'm a bit stuck at the moment now is how to actually display and represent the person and the ssh keys.

My thinking is either we can make a nice pretty-printer for ScimEntryGeneric, or we can deserialise into a stronger type for display. Not 100% sure yet.

**Comment by Firstyear** [comment]
This changes the ssh key list to the same format as today, but using the scim api. I think we need to do a bit of work on the cli in parallel to be ready for scim changes, I'm still just not sure how it should look. Either way, it solves the issue here which is an api to get ssh keys and labels cleanly.

**Review by CEbbinghaus — CHANGES_REQUESTED** [review]
Am I missing something? Where is the authorised_keys api?

File: proto/src/scim_v1/client.rs [hunk 1]
```diff
@@ -1 +1,11 @@
+use serde::{Deserialize, Serialize};
+use sshkey_attest::proto::PublicKey as SshPublicKey;
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply proto/src/scim_v1/client.rs:2]: PublicKey is ambiguous to other asymmetric public keys.

**Comment by Firstyear** [comment]
> Am I missing something? Where is the authorised_keys api?

For unixd, the keys are in the UnixUserToken, and then are provided to the caller. For ssh_authorized_keys_direct, it uses idm_account_get_ssh_pubkeys which calls https://github.com/kanidm/kanidm/blob/master/libs/client/src/lib.rs#L2021 instead of SCIM. In future, we should change this to use the SCIM variant.

**Comment by CEbbinghaus** [comment]
I got that. But both apis currently return json. The related issue asks for a endpoint that recieves the keys in a format compatible with the authorised_keys file. So you could do `curl [api] -o authorised_keys` to remove any post processing

**Comment by Firstyear** [comment]
"As an aside, it would also be nice to have an API that just returns the authorized_keys file's expected format, without JSON."

it's not the main issue of the ticket. 

Additionally, there can't easily be an api for this as we expect an auth bearer for requests, even for anonymous reqs.

**Review by CEbbinghaus — APPROVED** [review]
LGTM

**Review by yaleman — DISMISSED** [review]

**Review by yaleman — APPROVED** [review]

**Pull request merged**